sound/soc/sof/control.c

```diff
@@ -448,6 +450,11 @@ int snd_sof_bytes_ext_get(struct snd_kcontrol *kcontrol,
 
 	data_size = cdata->data->size + sizeof(const struct sof_abi_hdr);
 
+	/* limit size provided to user space if required */
+	/* FIXME: should we return an error instead ? */
```

Comment:
@plbossart I think its safer to return an error here unless we can assume that the user knows what they're doing

Reply:
also I think the bytes_volatile_get() function also needs to be changed to account for this

Reply:
both done.